File: 44_fizzbuzz/zig/main.zig
const std = @import("std");

pub fn main() !void {
    var args = std.process.args();

    // skips argv[0]
    _ = args.next();

    if (args.next()) |input| {
        const out = std.io.getStdOut().writer();

        const limit = try std.fmt.parseUnsigned(u16, input, 10);
        std.debug.print("Input: {d}\n", .{limit});

        for (0..limit) |n| {

            if (n % 15 == 0) {
                try out.print("{s}\n", .{"fizzbuzz"});
            } else if (n % 5 == 0) {
                try out.print("{s}\n", .{"fizz"});
            } else if (n % 3 == 0) {
                try out.print("{s}\n", .{"buzz"});
            } else {
                try out.print("{d}\n", .{n});
            }
        }

    } else {
        const out = std.io.getStdErr().writer();
        try out.print("error: {s}\n", .{"a numeric input is required"});
    }
}

File: 44_fizzbuzz/go/main.go
package main

import (
	"flag"
	"fmt"
)

func main() {

	limit := flag.Int("limit", 0, "Upper loop limit")
	flag.Parse()

	for i := 0; i < *limit; i++ {
		if i%15 == 0 {
			fmt.Printf("fizzbuzz\n")
		} else if i%5 == 0 {
			fmt.Printf("buzz\n")
		} else if i%3 == 0 {
			fmt.Printf("fizz\n")
		} else {
			fmt.Printf("%d\n", i)
		}
	}
}
